### DISCUSSION TITLE
Welcome to Sirius Web Discussions!

### DISCUSSION THREAD

**Discussion by pcdavid** [eclipse-sirius/sirius-components#2397] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by marekstasik** [comment]
Hello,
I have problem with SiriusWeb installation.  I use this instruction https://github.com/eclipse-sirius/sirius-web. I have problem with 4 step (Build the backend components).

![Screenshot from 2023-09-19 15-52-18](https://github.com/eclipse-sirius/sirius-web/assets/127620600/473a055e-5b53-4051-a015-157939fb2cd2)
![Screenshot from 2023-09-19 15-54-32](https://github.com/eclipse-sirius/sirius-web/assets/127620600/0cb4ed57-c7f8-40b4-837a-7eb1bf69bd0d)

**Reply by pcdavid** [reply]
Please create a separate thread/discussion for new topic.

Regarding this particular issue:
* which version of `sirius-web` are you using? current `master`?
* what build command are you invoking exactly? which versions of Java & Maven?

Sorry but we can't do much with screenshots of partial stack traces.

**Reply by pcdavid** [reply]
If the issue is only related to tests execution, try running Maven with `-DskipTests` and see if the build itself succeeds.

**Reply by pcdavid** [reply]
This is very probably related to Docker not being available.
I reproduce the error in a fresh VM (Fedora Desktop 38) with just Java & Maven installed (no Docker).

It is needed to execute some of the integration tests (to launch a containerized PostgreSQL instance).

Make sure you have Docker 1) installed, 2) running, 3) setup so that you can run it directly (i.e. not with `sudo`). You should be acble to execute `docker run hello-world` without error.
Check https://docs.docker.com/engine/install/ for installation instructions for your OS.

**Comment by marekstasik** [comment]
Thank you, it helped

**Comment by arlange** [comment]
Hi,

I have trouble to build the software. As soon I am trying to build the fronent I get an error:
<img width="1906" alt="Screenshot 2024-01-23 150606" src="https://github.com/eclipse-sirius/sirius-web/assets/118437561/c5153f15-75a2-4b29-b5e6-675534b55095">
I added the github access token to settings in my maven directory. 
OS: Ubuntu 22.04
Java: 17.0.9
Maven: 5.15.0-91

Is there another place where the access code is needed?

Regards Arne

**Reply by AxelRICHARD** [reply]
Hello, 
Please create a separate thread/discussion for new topic.
The problem is on the frontend side, so related to NPM, not Maven.
You should have a `.npmrc` in your user directory with the following content `//npm.pkg.github.com/:_authToken=INSERT_HERE_YOUR_PERSONAL_ACCESS_TOKEN`.
Regards,

**Reply by arlange** [reply]
I thought it would fit in here as well as the previous questions was related to building the software as well. Thanks for letting me know where to set access code as well.

**Reply by arlange** [reply]
.npmrc content: @ObeoNetwork:registry=https://npm.pkg.github.com/ObeoNetwork
that's it. 
added the line but error is still popping up.

**Reply by AxelRICHARD** [reply]
No, that's not what I suggested in my previous answer.
It should be `//npm.pkg.github.com/:_authToken=INSERT_HERE_YOUR_PERSONAL_ACCESS_TOKEN` where `INSERT_HERE_YOUR_PERSONAL_ACCESS_TOKEN` is the value of the personal access token you have generated from github, and you also use in your settings.xml file for maven.

**Reply by arlange** [reply]
I know that INSERT_HERE_YOUR_PERSONAL_ACCESS_TOKEN is just placeholder, I adjusted the line and added my personal access token for Github. Did not help.

**Reply by AxelRICHARD** [reply]
May be I'm wrong, but I think you mixed up the `.npmrc` file that is located in the sirius-web git repository and the `.npmrc` file that is located in your user directory (e.g. probably something like `C:/Users/arlange`). You don't have to modify the first one. If the second one doesn't exist, please create it with the content described above.
Regards,

**Reply by arlange** [reply]
Cool, it worked thanks.
